### PR TITLE
Support ReLU activation in CIndRnnLayer

### PR DESCRIPTION
**Pull request by FedyuninV**

### CODE REVIEW

File: NeoML/Python/neoml/Dnn/IndRnn.py [hunk 1]
```diff
@@ -56,7 +58,7 @@ class IndRnn(Layer):
         - **Channels** is hidden_size
     """
 
-    def __init__(self, input_layer, hidden_size=1, dropout_rate=0., reverse_sequence=False, name=None):
+    def __init__(self, input_layer, hidden_size=1, dropout_rate=0., reverse_sequence=False, activation='sigmoid', name=None):
```
Comment (SAngeliuk) [inline NeoML/Python/neoml/Dnn/IndRnn.py]: why sigmoid?

File: NeoML/include/NeoML/Dnn/Layers/IndRnnLayer.h [hunk 2]
```diff
@@ -115,14 +126,15 @@ class NEOML_API CIndRnnRecurrentLayer : public CBaseLayer {
 	void LearnOnce() override;
 
 private:
-	// int hiddenSize; // Number of elements in vector
+	IMathEngine::TIndRnnActivation activation; // Activation function
 	bool reverse; // If true then sequences must be processed in reversed order
 	float dropoutRate; // Dropout rate on recurrent link
-	CFloatHandleVar* dropoutMask; // Dropout mask
+	std::unique_ptr<CFloatHandleVar> dropoutMask; // Dropout mask
```
Comment (SAngeliuk) [inline NeoML/include/NeoML/Dnn/Layers/IndRnnLayer.h]: std here?

File: NeoMathEngine/include/NeoMathEngine/NeoMathEngine.h [hunk 3]
```diff
@@ -860,27 +860,36 @@ class NEOMATHENGINE_API IDnnEngine : public IBlasEngine {
...
+	//    h_t = activation( wx + u * h_(t-1))
 	// where
 	//    wx - user input (x), processed by fully connected layer (w). Size: seqLen x batchSize x objSize
 	//    mask - (optional, may be null) dropout mask. Size: batchSize x objSize
 	//    u - trainable vector of multipliers. Size: objSize
 
+	// Supported activations
+	// - sigmoid
+	// - ReLU
+	enum TIndRnnActivation {
```
Comment (SAngeliuk) [inline NeoMathEngine/include/NeoMathEngine/NeoMathEngine.h]: Lets use TActivationFunction!?

File: NeoML/include/NeoML/Dnn/Layers/IndRnnLayer.h (hunk 2 above)
Comment (FedyuninV) [reply NeoML/include/NeoML/Dnn/Layers/IndRnnLayer.h]: Why not? We already have `unique_ptr` in .cpp files in NeoML (e.g. BaseLayer.cpp)

File: NeoML/Python/neoml/Dnn/IndRnn.py (hunk 1 above)
Comment (FedyuninV) [reply NeoML/Python/neoml/Dnn/IndRnn.py]: fixed

File: NeoMathEngine/include/NeoMathEngine/NeoMathEngine.h (hunk 3 above)
Comment (FedyuninV) [reply NeoMathEngine/include/NeoMathEngine/NeoMathEngine.h]: fixed

File: NeoML/include/NeoML/Dnn/Layers/IndRnnLayer.h (hunk 2 above)
Comment (SAngeliuk) [reply NeoML/include/NeoML/Dnn/Layers/IndRnnLayer.h]: It is a header. We don't use std in headers...

Comment (FedyuninV) [reply NeoML/include/NeoML/Dnn/Layers/IndRnnLayer.h]: fixed

**Review by SAngeliuk — APPROVED** [review]

**Pull request merged**